lib/linguist/heuristics.rb

```diff
@@ -70,7 +71,7 @@ def matches?(filename, candidates)
     end
 
     # Internal: Perform the heuristic
-    def call(data)
+    def call(data, filename = nil)
       @heuristic.call(data)
```

Comment:
Are you sure this works? It's weird that `filename` is not used here, no?

Reply:
... I'm impressed at how I managed to overlook that, what the hell. 😂

I should really add a test for this.

Reply:
Will this suffice?

~~~rb
  def test_1to9_by_heuristics
    assert_heuristics({
      "Roff" => all_fixtures("Roff", "*.{1..9}"),
      "Text" => all_fixtures("Text", "*.{1..9}")
    })
  end
~~~